src/main/java/org/wso2/carbon/esb/module/ai/AbstractAIMediator.java

```diff
@@ -0,0 +1,137 @@
+
+    @Override
+    public void connect(MessageContext messageContext){
+        if (!initialized) {
+            initialize(messageContext);
+            initialized = true;
+        }
+        execute(messageContext);
+    }
+
+    protected <T> T getMediatorParameter(MessageContext messageContext, String parameterName, Class<T> type, boolean isOptional) {
+        Object parameter = getParameter(messageContext, parameterName);
+        if (!isOptional && (parameter == null || parameter.toString().isEmpty())) {
+            handleException(String.format("Parameter %s is not provided", parameterName), messageContext);
+        } else if (parameter == null || parameter.toString().isEmpty()) {
+            return null;
+        }
+
+        try {
+            return parse(Objects.requireNonNull(parameter).toString(), type);
+        } catch (IllegalArgumentException e) {
+            handleException(String.format("Parameter %s is not of type %s", parameterName, type.getName()), messageContext);
+        }
+
+        return null;
+    }
+
+    protected <T> T getProperty(MessageContext messageContext, String propertyName, Class<T> type, boolean isOptional) {
+        Object property = messageContext.getProperty(propertyName);
+        if (!isOptional && (property == null || property.toString().isEmpty())) {
+            handleException(String.format("Property %s is not set", propertyName), messageContext);
+        } else if (property == null || property.toString().isEmpty()) {
+            return null;
+        }
+
+        try {
+            return parse(Objects.requireNonNull(property).toString(), type);
+        } catch (IllegalArgumentException e) {
+            handleException(String.format("Property %s is not of type %s", propertyName, type.getName()), messageContext);
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T parse(String value, Class<T> type) throws IllegalArgumentException {
+        if (type == Integer.class) {
+            return (T) Integer.valueOf(value);
+        } else if (type == Double.class) {
+            return (T) Double.valueOf(value);
+        } else if (type == Boolean.class) {
+            return (T) Boolean.valueOf(value);
+        } else if (type == String.class) {
+            return (T) value;
+        } else {
+            throw new IllegalArgumentException("Unsupported type: " + type);
+        }
+    }
+
+    protected Object getObjetFromMC(MessageContext messageContext, String propertyName, boolean isOptional) {
+        Object property = messageContext.getProperty(propertyName);
+        if (property == null && !isOptional) {
+            handleException(String.format("Property %s is not set", propertyName), messageContext);
+        }
+        return property;
+    }
+
+    protected  void handleResponse(MessageContext messageContext, String responseVariable, Object payload, Map<String, Object> headers, Map<String, Object> attributes) {
```

Comment:
Lets check the length of the method here.

Reply:
fixed